File: WebSocketChat.Frontend/src/app/shared/models/chat/messages/server-information-message.ts
import { ChatRoom } from './../chat-room';
import { ChatUser } from './../chat-user';
import { Message, MessageIdentifier } from '../message';
import { MessageType } from '../message-type';

export class ServerInformationMessage extends MessageIdentifier implements Message {

	messageType : MessageType = MessageType.ServerInformation;

	/** A guid-to-chatpartner dictionary of all users which are connected to the server */
	userDictionary: { [key: string]: ChatUser };

	/** Dictionary of all available chatrooms  */
	availableRooms: { [key: number]: ChatRoom };
}

File: WebSocketChat.Server/WebSocketChat.Server/Models/Chat/Messages/ServerInformationMessage.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;

namespace WebSocketChat.Server.Models.Chat.Messages
{
    public class ServerInformationMessage : Message
    {
        public ServerInformationMessage()
        {
            MessageType = MessageType.ServerInformation;
        }

        public Dictionary<string, ChatPartner> UserDictionary { get; set; }

        public Dictionary<int, ChatRoom> AvailableRooms { get; set; }
    }
}
